### PR TITLE
New action that updates the reference in every recipe.

### PR DESCRIPTION
**Pull request by jsignell**
~~Oooh it almost works. Is it ok with you @forana if you have a bot that pushes straight to release branches like this?~~

This PR adds an action that opens a PR against release whenever a release branch is pushed to. I am not 100% sure if this will trigger when the branch is created, but if not I can change it.

### CODE REVIEW

**Comment by jsignell** [comment]
I realized that because of our branch protection policies, it needs to be in a PR.

**Comment by jsignell** [comment]
See https://github.com/saturncloud/examples/pull/183 for an example of this action working.

File: .ci/update-references.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+REF = sys.argv[1]
+
+
+for file in os.listdir("examples"):
+    recipe_path = os.path.join("examples", file, ".saturn/saturn.json")
+
+    with open(recipe_path) as f:
+        recipe = json.load(f)
+        for repo_dict in recipe["git_repositories"]:
+            repo_dict["reference"] = REF
```
Comment (jnolis) [inline .ci/update-references.py:14]: Will this require us to always include references in the recipes (as in we must include it even if it's `main`)? I think I am okay with that tradeoff I just want to make sure I understand correctly

**Review by jnolis — APPROVED** [review]
I had a minor comment but overall I think this is neat!

File: .github/workflows/update-references.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+name: update references
+on: create
+
+jobs:
+  update-reference:
+    if: ${{ contains(github.ref, 'refs/heads/release') }}
```
Comment (jsignell) [inline .github/workflows/update-references.yml:6]: Ok I updated this following this pattern: https://github.community/t/trigger-job-on-branch-created/16878/5 so I think it should happen on branch creation.

File: .ci/update-references.py (hunk 1 above)
Comment (jsignell) [reply .ci/update-references.py:14]: no it doesn't imply that. This just makes sure that we _do_ have references on release branches.

**Review by bhperry — APPROVED** [review]

**Pull request merged**